actor-tests/src/test/scala/org/apache/pekko/actor/SupervisorSpec.scala

```diff
@@ -95,6 +95,14 @@ object SupervisorSpec {
     }
   }
 
+  class Child(cnt: Int) extends Actor {
```

Comment:
seems like we need closing over the origin AtomicInteger because the restart will keep the actor state instead of recreate.

Reply:
I think this is ok... Because we were using creator.

https://github.com/apache/pekko/blob/46c557424d19bb2d07adf2fbd55351a8187ff359/actor/src/main/scala/org/apache/pekko/actor/Props.scala#L87-L88